src/games/strategy/triplea/ai/proAI/ProCombatMoveAI.java

```diff
@@ -48,147 +46,95 @@
-    final Map<Unit, Set<Territory>> bombardMap = new HashMap<Unit, Set<Territory>>();
-    final List<ProTransport> transportMapList = new ArrayList<ProTransport>();
+    data = ProData.getData();
+    player = ProData.getPlayer();
+    territoryManager = new ProTerritoryManager();
 
     // Determine whether capital is threatened and I should be in a defensive stance
-    isDefensive = !ProBattleUtils.territoryHasLocalLandSuperiority(myCapital, 3, player);
+    isDefensive = !ProBattleUtils.territoryHasLocalLandSuperiority(ProData.myCapital, 3, player);
     ProLogger.debug("Currently in defensive stance: " + isDefensive);
 
-    // Find all purchase options and min cost per hit point
-    final List<ProPurchaseOption> specialPurchaseOptions = new ArrayList<ProPurchaseOption>();
-    final List<ProPurchaseOption> factoryPurchaseOptions = new ArrayList<ProPurchaseOption>();
-    final List<ProPurchaseOption> landPurchaseOptions = new ArrayList<ProPurchaseOption>();
-    final List<ProPurchaseOption> airPurchaseOptions = new ArrayList<ProPurchaseOption>();
-    final List<ProPurchaseOption> seaPurchaseOptions = new ArrayList<ProPurchaseOption>();
-    ProPurchaseUtils.findPurchaseOptions(player, landPurchaseOptions, airPurchaseOptions, seaPurchaseOptions,
-        factoryPurchaseOptions, specialPurchaseOptions);
-    minCostPerHitPoint = ProPurchaseUtils.getMinCostPerHitPoint(player, landPurchaseOptions);
-
     // Find the maximum number of units that can attack each territory and max enemy defenders
-    final List<Territory> myUnitTerritories =
-        Match.getMatches(data.getMap().getTerritories(), Matches.territoryHasUnitsOwnedBy(player));
-    attackOptionsUtils.findAttackOptions(player, myUnitTerritories, attackMap, unitAttackMap, transportAttackMap,
-        bombardMap, transportMapList, new ArrayList<Territory>(), new ArrayList<Territory>(),
-        new ArrayList<Territory>(), false, false);
-    attackOptionsUtils.findScrambleOptions(player, attackMap);
-    final ProMoveOptions alliedAttackOptions = attackOptionsUtils.findAlliedAttackOptions(player);
-    final ProMoveOptions enemyDefendOptions = attackOptionsUtils.findEnemyDefendOptions(player);
+    territoryManager.populateAttackOptions();
+    territoryManager.populateEnemyDefenseOptions();
 
     // Remove territories that aren't worth attacking and prioritize the remaining ones
-    final List<ProTerritory> prioritizedTerritories =
-        attackOptionsUtils.removeTerritoriesThatCantBeConquered(player, attackMap, unitAttackMap, transportAttackMap,
-            alliedAttackOptions, enemyDefendOptions, false);
-    List<Territory> territoriesToAttack = new ArrayList<Territory>();
-    for (final ProTerritory patd : prioritizedTerritories) {
-      territoriesToAttack.add(patd.getTerritory());
+    final List<ProTerritory> attackOptions = territoryManager.removeTerritoriesThatCantBeConquered();
+    List<Territory> clearedTerritories = new ArrayList<Territory>();
```

Comment:
Yeah, looking to move to diamond notation all at once. I'll hold off on that one til the next round of changes.